internal/aws/awsutil/conn.go

```diff
@@ -83,7 +89,7 @@ func getProxyURL(finalProxyAddress string) (*url.URL, error) {
 }
 
 // GetAWSConfig returns AWS config instance.
-func GetAWSConfig(ctx context.Context, logger *zap.Logger, settings *AWSSessionSettings) (aws.Config, error) {
+func GetAWSConfig(ctx context.Context, logger *zap.Logger, settings *AWSSessionSettings, stsClient STSClient) (aws.Config, error) {
```

Comment:
It looks like you added the stsClient parameter so the client can be mocked, but you're only mocking it in this package.

I suggest you revert this one, and introduce a new unexported method for testing the STS part within this package. That would keep the API clean for the exporters.